nifi-nar-bundles/nifi-pulsar-bundle/nifi-pulsar-client-service/src/main/java/org/apache/nifi/pulsar/StandardPulsarClientService.java

```diff
@@ -0,0 +1,303 @@
+            .name("LISTENER_THREADS")
+            .displayName("Listener Threads")
+            .description("The number of threads to be used for message listeners")
+            .required(false)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .defaultValue("1")
+            .build();
+
+    public static final PropertyDescriptor MAXIMUM_LOOKUP_REQUESTS = new PropertyDescriptor.Builder()
+            .name("MAXIMUM_LOOKUP_REQUESTS")
+            .displayName("Maximum lookup requests")
+            .description("Number of max lookup-requests allowed on each broker-connection. To prevent overload on broker, "
+                       + "it should be greater than the 'Maximum concurrent lookup-requests' property value.")
+            .required(false)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("50000")
+            .build();
+
+    public static final PropertyDescriptor MAXIMUM_REJECTED_REQUESTS = new PropertyDescriptor.Builder()
+            .name("MAXIMUM_REJECTED_REQUESTS")
+            .displayName("Maximum rejected requests per connection")
+            .description("Max number of broker-rejected requests in a certain time-frame after " +
+                    "which current connection will be closed and client creates a new connection that gives " +
+                    "chance to connect a different broker.")
+            .required(false)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .defaultValue("50")
+            .build();
+
+    public static final PropertyDescriptor OPERATION_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("OPERATION_TIMEOUT")
+            .displayName("Operation Timeout")
+            .description("Producer-create, subscribe and unsubscribe operations will be retried until this " +
+                    "interval, after which the operation will be marked as failed.")
+            .required(false)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .defaultValue("30 sec")
+            .build();
+
+    public static final PropertyDescriptor STATS_INTERVAL = new PropertyDescriptor.Builder()
+            .name("STATS_INTERVAL")
+            .displayName("Stats interval")
+            .description("The interval between each stat infomation update. It should be set to at least 1 second.")
+            .required(false)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .defaultValue("60 sec")
+            .build();
+
+    public static final PropertyDescriptor TLS_TRUST_CERTS_FILE_PATH = new PropertyDescriptor.Builder()
```

Comment:
Given the SSLContext below what is this necessary for?  All the TLS establishment details should come from the SSL Context.

Reply:
I will remove this property as well